File: CursoIWTraining/Advanced/AcessoWeb/app/src/main/java/com/dvidal/acessoweb/MainActivity.java
package com.dvidal.acessoweb;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;

import java.util.List;

public class MainActivity extends AppCompatActivity implements View.OnClickListener, StudentService.StudentServiceListener {

    private Button btnConnect;
    private Spinner spinnerStudents;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        btnConnect = (Button)findViewById(R.id.btnConnect);

        btnConnect.setOnClickListener(this);

        spinnerStudents = (Spinner) findViewById(R.id.spinnerStudents);

    }


    @Override
    public void onClick(View v) {
        if (Connection.verifyConnection(this)){

            StudentService ss = new StudentService();
            ss.setOnStudentServiceListener(this);
        }

    }

    @Override
    public void onStudentServiceSucessfully(List<Student> students) {
        Log.d("General", "Deu certo!!");

        ArrayAdapter adapter = new ArrayAdapter(this, R.layout.support_simple_spinner_dropdown_item, students);
        spinnerStudents.setAdapter(adapter);
    }

    @Override
    public void onStudentServiceFailure(String error) {
        Log.d("General", error);
    }
}

File: fragment-mvvm-rxjava2-dagger2-example/app/src/main/java/br/com/vp/advancedandroid/home/MainActivity.kt
package br.com.vp.advancedandroid.home

import android.support.v4.app.Fragment
import br.com.vp.advancedandroid.R
import br.com.vp.advancedandroid.base.BaseActivity
import br.com.vp.advancedandroid.trending.TrendingReposFragment

class MainActivity : BaseActivity() {

    override fun layoutRes(): Int {

        return R.layout.activity_main
    }

    override fun initialScreen(): Fragment {

        return TrendingReposFragment()
    }
}
